src/components/MelodyDetailsPage.js

```diff
@@ -0,0 +1,120 @@
+import React from "react";
+import ReactAbcjs from "./react-abcjs.js";
+import MusicInputFormContainer from "./MusicInputFormContainer";
+import "./MelodyDetailsPage.css";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faPlayCircle } from "@fortawesome/free-solid-svg-icons";
+
+function MelodyDetailsPage(props) {
+  let replacedText = "";
+  if (props.melody) {
+    replacedText = props.melody.abcStart.replace(/Q:.*\nI:.*\n/, "");
```

Comment:
I don't know how often this component re-renders.
But it good way to calculate some vars one time if props for it not changed.
```
const replacedText = useMemo(() => props.melody ? props.melody.abcStart.replace(/Q:.*\nI:.*\n/, "") : "", [props.melody])
```
More about that: https://reactjs.org/docs/hooks-reference.html#usememo